libraries/rig-package/package.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@rushstack/rig-package",
+  "version": "0.0.0",
+  "description": "A standard for sharing tool configurations between projects without duplicating config files.",
+  "main": "lib/index.js",
+  "typings": "dist/rig-package.d.ts",
+  "license": "MIT",
+  "repository": {
+    "url": "https://github.com/microsoft/rushstack/tree/master/libraries/rig-package"
+  },
+  "scripts": {
+    "build": "heft test --clean"
+  },
+  "dependencies": {
```

Comment:
I don't think making `resolve` be a dependency of this project would be a problem, and would make its usage way simpler.

Right now, anyone who uses this package has to:

1. Take their own dependency on [a potentially unstable version of] `resolve` and pass it in (what almost everyone is going to do)
2. Use NCL (which is what we're going to do, and which has a dependency on a stable version of `resolve`)
3. Write their own resolver (let's be honest, nobody sane is going to do this)

I understand the "purity" of having no dependencies, but I really think taking a dependency on `resolve` will make this package easier to use and will allow us to provide an async method for resolving the rig package (in addition to a sync one, if you want to retain that)